### ISSUE TITLE
Cannot publish a schema in an issuer agent when turning on the multi-tenancy mode

### ISSUE THREAD

**Issue by yunxi-zhang** [hyperledger/aries-cloudagent-python#1158]
I have an agent to run as issuers by turning on the multi-tenancy mode. At the agent level, i've registered a public DID and I can see this public DID information shown in the agent log when it runs.

![image](https://user-images.githubusercontent.com/16185797/117301516-9ee14400-ae72-11eb-9066-73600cf6a9cb.png)

I then create 2 sub wallets in this issuer agent. When I use sub wallet 1 to publish a new credential schema, I've got below errors. When I check the DIDs held by sub wallet 1, there is no public DID inside. 

> 400: Cannot publish schema without a public DID.

Question: does it mean an issuer agent can't use multi-tenancy mode? If answer is no, how should I add public DIDs to each sub wallet in the issuer agent?

**Comment by esune** [comment]
You will need to first create a did using `/wallet/did/create` and publish it using `/wallet/did/public?did={myPublicDID}`.

Also be aware that you will need to register the new DID on the ledger as you normally would do for any other agent before it can perform writes.

An example of an issuer agency (focused on issuing to Aries VCR) is [here](https://github.com/bcgov/aries-vcr-issuer-agency) (still under development at the time of writing).

**Comment by yunxi-zhang** [comment]
Hello @esune, i've done the below steps
1. create a local did by using api: `/wallet/did/create`
2. register this did on the von network browser to write this did to the ledger, see below picture
![image](https://user-images.githubusercontent.com/16185797/117466135-49786600-af4a-11eb-9de4-c1ecda1cfbe8.png)
3. call publish did api: `/wallet/did/public?did=B6MJpTFHWTSoNThFp6w8Pc`, and i've got below error:
`400: Ledger rejected transaction request: client request invalid: insufficient number of valid signatures, 1 is required but 0 valid and 1 invalid have been provided. The following signatures are invalid: did=B6MJpTFHWTSoNThFp6w8Pc, signature=2pp5kWgdCT8HgR4zR7dTAApKcgEhr2BWKYC7M36bBCUxQFr562UCnvWcnZesd2gxmN3EsmdYxf8Qg9fiFirKBXUe. `

Could you give a help on how to fix it?

**Comment by esune** [comment]
I will try and reproduce the issue and get back to you

**Comment by ianco** [comment]
Hi @yunxi-zhang it sounds like your issues are with von-network rather than aries-cloudagent.

Can you try:  shut down all your docker containers (`./manage rm` in all shells) and then just start your von-network (`./manage start --logs`).  Try creating a DID from a seed (you just need a 32 character seed) and confirm whether that works.  If not can you post the full set of logs from von-network, as well as info about the platform you are running on?

If creating a DID from a seed works, try creating the same DID and verkey as in the errir above, again if you get an error please post the full set of logs etc.

If this works, then try starting aca-py again.

Another thing to try - try running the alice/faber demo in multitenant mode (`./run_demo faber --multitenant`), you can create a new sub-wallet and issue a credential using this demo.  You can review what the demo code is doing vs what you are doing when you test manually and see if there are an y differences.

**Comment by esune** [comment]
@yunxi-zhang I have not had a chance to troubleshoot too in depth, and was unable to replicate the issue s far.

I think @ianco is right and you might have an issue with your `von-network` and having the nodes reach consensus maybe? Would you mind trying what he recommends and reporting back?

**Comment by yunxi-zhang** [comment]
@ianco , thanks for the guidance, publishing a DID for a wallet works this time. The only difference I've done this time compared to what I did previously is I put a 32 characters (issuer00000000000000000000000000) for seed when using "registering from seed", whilst previously I only put 6 characters (issuer). Not sure if this difference caused the issue.

**Comment by esune** [comment]
> 
> 
> @ianco , thanks for the guidance, publishing a DID for a wallet works this time. The only difference I've done this time compared to what I did previously is I do put a 32 characters (issuer00000000000000000000000000) for seed when using "registering from seed" this time, whilst previously I only put 6 characters (issuer). Not sure if this difference caused the issue.

Yes, the seed must be a 32 character string. I am surprised a more explicit message was not raised - it's great that you managed to sort it out!

**Comment by yunxi-zhang** [comment]
@esune , wonder if it's worthing adding some logics in von-network when users not giving 32 character strings. Because when the multi-tenancy mode is not enabled, I always give less than 32 characters, and it always works.

**Comment by esune** [comment]
I do not think this is a multi-tenancy related setting, it should always be a 32-character seed (see screenshot from running von-network instance). I am actually surprised you can register seeds that are less than 32 characters.
![image](https://user-images.githubusercontent.com/2395873/118007371-f06d5100-b300-11eb-9d9a-70cb83bc5347.png)

**Comment by swcurran** [comment]
I suspect that some methods of adding a seed do padding of the seed if you don't provide the 32 characters.  It's not very safe, as you don't really control what they do to pad the seed -- do they add spaces? Zeroes? Ones? Much better to do it yourself, but for PoCs that are using the same method every time it's OK.

For production uses, you should not use these techniques, but only something local on your own machine, and save and safely preserve the seed.
